### PR TITLE
regularize constructor's uid when parsing url and anchor

### PR DESCRIPTION
**Pull request by ansyral**
#503

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs [hunk 1]
```diff
@@ -41,16 +41,16 @@ public static XRefDetails From(HtmlAgilityPack.HtmlNode node)
             var rawUid = GetRawUid(node);
             if (!string.IsNullOrEmpty(rawUid))
             {
-                var anchorIndex = rawUid.IndexOf("#");
+                var anchorIndex = rawUid.Replace(".#ctor", "..ctor").IndexOf("#");
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs]: what about EII?

**Comment by vwxyzh** [comment]
✔️

File: src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs [hunk 2]
```diff
@@ -41,16 +41,16 @@ public static XRefDetails From(HtmlAgilityPack.HtmlNode node)
...
                 if (anchorIndex == -1)
                 {
                     xref.Anchor = string.Empty;
-                    xref.Uid = HttpUtility.UrlDecode(rawUid);
+                    xref.Uid = HttpUtility.UrlDecode(rawUid.Replace("..ctor", ".#ctor"));
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs]: >.Replace("..ctor", ".#ctor")) [](start = 59, length = 29)

No need to replace here? The rawUID doesn't change after Replace().

File: src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs [hunk 3]
```diff
@@ -41,16 +41,16 @@ public static XRefDetails From(HtmlAgilityPack.HtmlNode node)
...
                 }
                 else
                 {
                     xref.Anchor = rawUid.Substring(anchorIndex);
-                    xref.Uid = HttpUtility.UrlDecode(rawUid.Remove(anchorIndex));
+                    xref.Uid = HttpUtility.UrlDecode(rawUid.Remove(anchorIndex).Replace("..ctor", ".#ctor"));
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs]: >.Replace("..ctor", ".#ctor") [](start = 79, length = 28)

no need?

File: src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs (hunk 2 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs]: what about System.String#System.String.#ctor

File: src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs (hunk 3 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs]: what about System.String#System.String.#ctor

File: src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs (hunk 2 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs]: `.#ctor` was replaced with `..ctor`  before

File: src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs (hunk 1 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/XrefDetails.cs]: good question

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs [hunk 4]
```diff
@@ -283,7 +284,7 @@ private void ResolveCrefLink(XNode node, string nodeSelector, Action<string, str
                         // When see and seealso are top level nodes in triple slash comments, do not convert it into xref node
                         if (item.Parent?.Parent != null)
                         {
-                            var replacement = XElement.Parse($"<xref href=\"{WebUtility.HtmlEncode(value)}\" data-throw-if-not-resolved=\"false\"></xref>");
+                            var replacement = XElement.Parse($"<xref href=\"{HttpUtility.UrlEncode(value)}\" data-throw-if-not-resolved=\"false\"></xref>");
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs]: >HttpUtility.UrlEncode [](start = 77, length = 21)

Uri.EscapeDataString

**Comment by chenkennt** [comment]
:+1:

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:+1:

**Comment by qinezh** [comment]
:100:

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs (hunk 4 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Parsers/TripleSlashCommentModel.cs]: pending. need to check places that used UrlDecode and change as a whole.

**Pull request merged**